packages/parser/src/components/parser/statement/expression/callableExpression.ts

```diff
@@ -0,0 +1,34 @@
+import Expression from ".";
+
+import { TokenTypes } from "../../../../constants/bhaiLangSpec";
+import { NodeType } from "../../../../constants/constants";
+import { ASTNode } from "../../types/nodeTypes";
+
+export default class CallableExpression extends Expression {
+	getExpression(): ASTNode {
+		const name = this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.CALLABLE_TYPE).value;
+		this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.OPEN_PARENTHESIS_TYPE);
+
+		// read arguments
+		let args: any[] = [];
+		if(this._tokenExecutor.getLookahead()?.type !== TokenTypes.CLOSED_PARENTHESIS_TYPE) {
+			do {
+				args.push(this._getArgs());
+			} while (
+				this._tokenExecutor.getLookahead()?.type === TokenTypes.COMMA_TYPE && 
+				this._tokenExecutor.getLookahead()?.type !== TokenTypes.CLOSED_PARENTHESIS_TYPE&&
+				this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.COMMA_TYPE)
+			);
+		}
+		
+		this._tokenExecutor.eatTokenAndForwardLookahead(TokenTypes.CLOSED_PARENTHESIS_TYPE);
+		return {
+			type: NodeType.CallableExpression,
+			name,
+			args,
+		};
+	}
+	private _getArgs() {
+		return Expression.getExpressionImpl(NodeType.AssignmentExpression).getExpression();
```

Comment:
I am skeptical about using assignment expression here. 

I would like to see the output of this function declaration:

```

func(a+b) {
......something
}

```

please cover this also in the test cases if not covered already.

Reply:
"callable expression" is not during function declaration but during the function call, and `func(1+2)` is a valid expression in that scenario

Reply:
agreed, that was an oversight, I changed the code in the `function statement`. but the code highlighted here is for callable expression, which is used for calling functions or classes (in future) example not for writing the definitions
```
bol bhai  square(10+20);
```